ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapTypeDialogFragment.kt

```diff
@@ -59,6 +59,11 @@ class MapTypeDialogFragment : Hilt_MapTypeDialogFragment() {
     val recyclerview = binding.recyclerView
     recyclerview.adapter =
       MapTypeAdapter(requireContext(), mapTypes, index) { onMapTypeSelected(it) }
+
+    binding.offlineMapPreferenceSwitch.isChecked = mapStateRepository.isOfflineImageryEnabled
+    binding.offlineMapPreferenceSwitch.setOnCheckedChangeListener { _, isChecked ->
```

Comment:
Can we instead add an action to the layout to update the state via its ViewModel?

Reply:
Done